File: go/example_code/ec2/ec2_delete_keypair.go
// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: Apache-2.0

package main

import (
	"fmt"
	"os"
	"path/filepath"

	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/awserr"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/ec2"
)

// Deletes a EC2 key pair for the name provided. No error will be returned
// if the key pair does not exist.
//
// Usage:
//
//	go run ec2_delete_keypair.go KEY_PAIR_NAME
func main() {
	if len(os.Args) != 2 {
		exitErrorf("pair name required\nUsage: %s key_pair_name",
			filepath.Base(os.Args[0]))
	}
	pairName := os.Args[1]

	// Initialize a session in us-west-2 that the SDK will use to load
	// credentials from the shared credentials file ~/.aws/credentials.
	sess, err := session.NewSession(&aws.Config{
		Region: aws.String("us-west-2")},
	)

	// Create an EC2 service client.
	svc := ec2.New(sess)

	// Delete the key pair by name
	_, err = svc.DeleteKeyPair(&ec2.DeleteKeyPairInput{
		KeyName: aws.String(pairName),
	})
	if err != nil {
		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == "InvalidKeyPair.Duplicate" {
			exitErrorf("Key pair %q does not exist.", pairName)
		}
		exitErrorf("Unable to delete key pair: %s, %v.", pairName, err)
	}

	fmt.Printf("Successfully deleted %q key pair\n", pairName)
}

func exitErrorf(msg string, args ...interface{}) {
	fmt.Fprintf(os.Stderr, msg+"\n", args...)
	os.Exit(1)
}

File: cpp/example_code/ec2/ec2_delete_key_pair.cpp
/*
   Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.

   This file is licensed under the Apache License, Version 2.0 (the "License").
   You may not use this file except in compliance with the License. A copy of
   the License is located at

    http://aws.amazon.com/apache2.0/

   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
   CONDITIONS OF ANY KIND, either express or implied. See the License for the
   specific language governing permissions and limitations under the License.
*/
#include <aws/core/Aws.h>

#include <aws/ec2/EC2Client.h>
#include <aws/ec2/model/DeleteKeyPairRequest.h>

#include <iostream>

/**
 * Deletes an ec2 key pair based on command line input
 */
int main(int argc, char** argv)
{
    if(argc != 2)
    {
        std::cout << "Usage: ec2_delete_key_pair <key_pair_name>" << std::endl;
        return 1;
    }

    Aws::String keyPairName = argv[1];

    Aws::SDKOptions options;
    Aws::InitAPI(options);

    {
        Aws::EC2::EC2Client ec2_client;

        Aws::EC2::Model::DeleteKeyPairRequest deleteKeyPairRequest;
        deleteKeyPairRequest.SetKeyName(keyPairName);

        auto deleteKeyPairOutcome = ec2_client.DeleteKeyPair(deleteKeyPairRequest);
        if (!deleteKeyPairOutcome.IsSuccess())
        {
            std::cout << "Failed to delete key pair " << keyPairName << ":" <<
            deleteKeyPairOutcome.GetError().GetMessage() << std::endl;
        }
        else
        {
            std::cout << "Successfully deleted key pair named " << keyPairName << std::endl;
        }
    }

    Aws::ShutdownAPI(options);

    return 0;
}



